crates/block-producer/src/mempool/mod.rs

```diff
@@ -198,28 +199,45 @@ pub struct Mempool {
     /// The current inflight block, if any.
     block_in_progress: Option<BTreeSet<BatchJobId>>,
 
-    batch_budget: BatchBudget,
+    /// Batches which are currently being proven.
+    ///
+    /// This is used to identify jobs which have been cancelled by the mempool but might still be
+    /// submitted by the batch prover. This is achieved by ignoring all batch proofs which are not
+    /// in this set.
+    batches_in_progress: BTreeSet<BatchJobId>,
```

Comment:
Would it make sense to track which batches are in-progress inside of the `BatchGraph`? If we need to access this info here, we could also add `BatchGraph::isInProgress(BatchJobId)` method. Or would this not work for some reason?

Reply:
I've removed this entirely, and instead check that the batch is contained within the graph beforehand instead - this is equivalent.